diag_manager/fms_diag_yaml.F90

```diff
@@ -1,62 +1,328 @@
+
+!< @brief gets the lat/lon of the sub region to use in a diag_table yaml
+subroutine get_sub_region_latlon(diag_yaml_id, sub_region_id, lat_lon_sub_region)
+  integer, intent(in)  :: diag_yaml_id       !< Id of the diag_table yaml file
+  integer, intent(in)  :: sub_region_id      !< Id of the region block to read from
+  real,    intent(out) :: lat_lon_sub_region (NUM_SUB_REGION_ARRAY) !< Array storing the bounds of the sub region
+
+  call get_value_from_key(diag_yaml_id, sub_region_id, "dim1_begin", lat_lon_sub_region(1), is_optional=.true.)
+  call get_value_from_key(diag_yaml_id, sub_region_id, "dim1_end", lat_lon_sub_region(2), is_optional=.true.)
+  call get_value_from_key(diag_yaml_id, sub_region_id, "dim2_begin", lat_lon_sub_region(3), is_optional=.true.)
+  call get_value_from_key(diag_yaml_id, sub_region_id, "dim2_end", lat_lon_sub_region(4), is_optional=.true.)
+  call get_value_from_key(diag_yaml_id, sub_region_id, "dim3_begin", lat_lon_sub_region(5), is_optional=.true.)
+  call get_value_from_key(diag_yaml_id, sub_region_id, "dim3_end", lat_lon_sub_region(6), is_optional=.true.)
+  call get_value_from_key(diag_yaml_id, sub_region_id, "dim4_begin", lat_lon_sub_region(7), is_optional=.true.)
+  call get_value_from_key(diag_yaml_id, sub_region_id, "dim4_end", lat_lon_sub_region(8), is_optional=.true.)
+
+end subroutine get_sub_region_latlon
+
+!< @brief gets the indexes of the sub region to use in a diag_table yaml
+subroutine get_sub_region_index(diag_yaml_id, sub_region_id, index_sub_region)
+  integer, intent(in)  :: diag_yaml_id     !< Id of the diag_table yaml file
+  integer, intent(in)  :: sub_region_id    !< Id of the region block to read from
+  integer, intent(out) :: index_sub_region (NUM_SUB_REGION_ARRAY) !< Array storing the index bounds of the sub region
+
+  call get_value_from_key(diag_yaml_id, sub_region_id, "dim1_begin", index_sub_region(1), is_optional=.true.)
+  call get_value_from_key(diag_yaml_id, sub_region_id, "dim1_end", index_sub_region(2), is_optional=.true.)
+  call get_value_from_key(diag_yaml_id, sub_region_id, "dim2_begin", index_sub_region(3), is_optional=.true.)
+  call get_value_from_key(diag_yaml_id, sub_region_id, "dim2_end", index_sub_region(4), is_optional=.true.)
+  call get_value_from_key(diag_yaml_id, sub_region_id, "dim3_begin", index_sub_region(5), is_optional=.true.)
+  call get_value_from_key(diag_yaml_id, sub_region_id, "dim3_end", index_sub_region(6), is_optional=.true.)
+  call get_value_from_key(diag_yaml_id, sub_region_id, "dim4_begin", index_sub_region(7), is_optional=.true.)
+  call get_value_from_key(diag_yaml_id, sub_region_id, "dim4_end", index_sub_region(8), is_optional=.true.)
+
+end subroutine get_sub_region_index
+
+!< @brief gets the total number of variables in the diag_table yaml file
+!< @return total number of variables
+function get_total_num_vars(diag_yaml_id, file_ids) &
+result(total_nvars)
+
+  integer, intent(in) :: diag_yaml_id   !< Id for the diag_table yaml
```

Comment:
I think you want this to be `diag_file_id` or `history_file_id`.  I thought this was the fortran unit number for the diag_table.yaml file.